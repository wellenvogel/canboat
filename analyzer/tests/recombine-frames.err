ERROR recombine [analyzer] Received incomplete fast packet PGN 129029 from source 0
