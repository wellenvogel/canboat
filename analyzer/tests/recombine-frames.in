# Multiple frames, in order, with some other short frame interspersed
# Should result in 2 explained PGNs
2022-09-28-11:36:59.668,3,129029,0,255,8,00,2f,e7,95,3d,00,73,d6
2022-09-28-11:36:59.668,3,129029,0,255,8,01,29,00,da,04,73,db,c9
2022-09-28-11:36:59.668,3,129029,0,255,8,02,e5,05,80,7d,02,28,5f
2022-09-28-11:36:59.668,3,129029,0,255,8,03,d6,10,f6,9b,50,6c,05
2022-09-28-11:36:59.668,5,130311,35,255,8,c5,c0,1c,6e,ff,7f,ff,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,04,00,00,00,00,13,fc,08
2022-09-28-11:36:59.668,3,129029,0,255,8,05,6f,00,be,00,dd,f2,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,06,ff,00,ff,ff,ff,ff,ff

# Multiple frames, with 2 different PGNs from same source interspersed
# Should result in 2 explained PGNs
2022-09-28-11:36:59.668,3,129029,0,255,8,00,2f,e7,95,3d,00,73,d6
2022-09-28-11:36:59.668,3,129029,0,255,8,01,29,00,da,04,73,db,c9
2022-09-28-11:36:59.669,7,126720,0,255,8,20,3e,e5,98,17,00,04,04
2022-09-28-11:36:59.669,7,126720,0,255,8,21,bf,a0,1b,41,5e,14,7f
2022-09-28-11:36:59.668,3,129029,0,255,8,02,e5,05,80,7d,02,28,5f
2022-09-28-11:36:59.668,3,129029,0,255,8,03,d6,10,f6,9b,50,6c,05
2022-09-28-11:36:59.669,7,126720,0,255,8,22,41,4c,67,95,41,4c,67
2022-09-28-11:36:59.669,7,126720,0,255,8,23,95,41,4c,67,95,41,0a
2022-09-28-11:36:59.668,3,129029,0,255,8,04,00,00,00,00,13,fc,08
2022-09-28-11:36:59.668,3,129029,0,255,8,05,6f,00,be,00,dd,f2,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,06,ff,00,ff,ff,ff,ff,ff
2022-09-28-11:36:59.669,7,126720,0,255,8,24,d7,a3,3c,cd,cc,cc,3d
2022-09-28-11:36:59.669,7,126720,0,255,8,25,0a,d7,a3,3c,cd,cc,cc
2022-09-28-11:36:59.669,7,126720,0,255,8,26,3d,a4,17,8e,3f,e6,e1
2022-09-28-11:36:59.669,7,126720,0,255,8,27,c5,3f,23,9d,f3,3f,00
2022-09-28-11:36:59.669,7,126720,0,255,8,28,00,80,3f,2b,34,84,3e

# Multiple frames, out of order.
# Should result in 1 explained PGN
2022-09-28-11:36:59.668,3,129029,0,255,8,00,2f,e7,95,3d,00,73,d6
2022-09-28-11:36:59.668,3,129029,0,255,8,02,e5,05,80,7d,02,28,5f
2022-09-28-11:36:59.668,3,129029,0,255,8,06,ff,00,ff,ff,ff,ff,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,03,d6,10,f6,9b,50,6c,05
2022-09-28-11:36:59.668,3,129029,0,255,8,04,00,00,00,00,13,fc,08
2022-09-28-11:36:59.668,3,129029,0,255,8,05,6f,00,be,00,dd,f2,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,01,29,00,da,04,73,db,c9

# Multiple frames, out of order, not all frames transmitted, followed by another out-of-order.
# Should result in 1 explained PGN
2022-09-28-11:36:59.668,3,129029,0,255,8,00,2f,e7,95,3d,00,73,d6
2022-09-28-11:36:59.668,3,129029,0,255,8,02,e5,05,80,7d,02,28,5f
2022-09-28-11:36:59.668,3,129029,0,255,8,06,ff,00,ff,ff,ff,ff,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,03,d6,10,f6,9b,50,6c,05
2022-09-28-11:36:59.668,3,129029,0,255,8,04,00,00,00,00,13,fc,08
2022-09-28-11:36:59.668,3,129029,0,255,8,05,6f,00,be,00,dd,f2,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,02,e5,05,80,7d,02,28,5f
2022-09-28-11:36:59.668,3,129029,0,255,8,00,2f,e7,95,3d,00,73,d6
2022-09-28-11:36:59.668,3,129029,0,255,8,06,ff,00,ff,ff,ff,ff,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,03,d6,10,f6,9b,50,6c,05
2022-09-28-11:36:59.668,3,129029,0,255,8,04,00,00,00,00,13,fc,08
2022-09-28-11:36:59.668,3,129029,0,255,8,05,6f,00,be,00,dd,f2,ff
2022-09-28-11:36:59.668,3,129029,0,255,8,01,29,00,da,04,73,db,c9
