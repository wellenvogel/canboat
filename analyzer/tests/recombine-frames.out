2022-09-28-11:36:59.668 5  35 255 130311 Environmental Parameters:  SID = 197 (bytes = "C5"); Temperature Source = Sea Temperature (bytes = "00", bits = "000000"); Humidity Source = Unknown (bytes = "C0", bits = "11"); Temperature = 8.73 C (bytes = "1C 6E"); Humidity = 131.068 % (bytes = "FF 7F"); Atmospheric Pressure = Unknown (bytes = "FF FF")
2022-09-28-11:36:59.668 3   0 255 129029 GNSS Position Data:  SID = 231 (bytes = "E7"); Date = 2013.03.01 (bytes = "95 3D"); Time = 19:29:52 (bytes = "00 73 D6 29"); Latitude = 42.4967684 (bytes = "00 DA 04 73 DB C9 E5 05"); Longitude = -71.5836637 (bytes = "80 7D 02 28 5F D6 10 F6"); Altitude = 90.984603 m (bytes = "9B 50 6C 05 00 00 00 00"); GNSS type = GPS+SBAS/WAAS (bytes = "03", bits = "0011"); Method = GNSS fix (bytes = "10", bits = "0001"); Integrity = No integrity checking (bytes = "00", bits = "00"); Number of SVs = 8 (bytes = "08"); HDOP = 1.11 (bytes = "6F 00"); PDOP = 1.90 (bytes = "BE 00"); Geoidal Separation = -33.63 m (bytes = "DD F2 FF FF"); Reference Stations = 0 (bytes = "00"); Reference Station Type 1 = Unknown (bytes = "0F", bits = "1111"); Reference Station ID = Unknown (bytes = "F0 FF", bits = "111111111111"); Age of DGNSS Corrections = Unknown (bytes = "FF FF")
2022-09-28-11:36:59.668 3   0 255 129029 GNSS Position Data:  SID = 231 (bytes = "E7"); Date = 2013.03.01 (bytes = "95 3D"); Time = 19:29:52 (bytes = "00 73 D6 29"); Latitude = 42.4967684 (bytes = "00 DA 04 73 DB C9 E5 05"); Longitude = -71.5836637 (bytes = "80 7D 02 28 5F D6 10 F6"); Altitude = 90.984603 m (bytes = "9B 50 6C 05 00 00 00 00"); GNSS type = GPS+SBAS/WAAS (bytes = "03", bits = "0011"); Method = GNSS fix (bytes = "10", bits = "0001"); Integrity = No integrity checking (bytes = "00", bits = "00"); Number of SVs = 8 (bytes = "08"); HDOP = 1.11 (bytes = "6F 00"); PDOP = 1.90 (bytes = "BE 00"); Geoidal Separation = -33.63 m (bytes = "DD F2 FF FF"); Reference Stations = 0 (bytes = "00"); Reference Station Type 1 = Unknown (bytes = "0F", bits = "1111"); Reference Station ID = Unknown (bytes = "F0 FF", bits = "111111111111"); Age of DGNSS Corrections = Unknown (bytes = "FF FF")
2022-09-28-11:36:59.669 7   0 255 126720 0x1F000-0x1FEFF: Standardized mixed single/fast packet non-addressed:  Data = E5 98 17 00 04 04 BF A0 1B 41 5E 14 7F 41 4C 67 95 41 4C 67 95 41 4C 67 95 41 0A D7 A3 3C CD CC CC 3D 0A D7 A3 3C CD CC CC 3D A4 17 8E 3F E6 E1 C5 3F 23 9D F3 3F 00 00 80 3F 2B 34 84 3E (bytes = "E5 98 17 00 04 04 BF A0 1B 41 5E 14 7F 41 4C 67 95 41 4C 67 95 41 4C 67 95 41 0A D7 A3 3C CD CC CC 3D 0A D7 A3 3C CD CC CC 3D A4 17 8E 3F E6 E1 C5 3F 23 9D F3 3F 00 00 80 3F 2B 34 84 3E")
2022-09-28-11:36:59.668 3   0 255 129029 GNSS Position Data:  SID = 231 (bytes = "E7"); Date = 2013.03.01 (bytes = "95 3D"); Time = 19:29:52 (bytes = "00 73 D6 29"); Latitude = 42.4967684 (bytes = "00 DA 04 73 DB C9 E5 05"); Longitude = -71.5836637 (bytes = "80 7D 02 28 5F D6 10 F6"); Altitude = 90.984603 m (bytes = "9B 50 6C 05 00 00 00 00"); GNSS type = GPS+SBAS/WAAS (bytes = "03", bits = "0011"); Method = GNSS fix (bytes = "10", bits = "0001"); Integrity = No integrity checking (bytes = "00", bits = "00"); Number of SVs = 8 (bytes = "08"); HDOP = 1.11 (bytes = "6F 00"); PDOP = 1.90 (bytes = "BE 00"); Geoidal Separation = -33.63 m (bytes = "DD F2 FF FF"); Reference Stations = 0 (bytes = "00"); Reference Station Type 1 = Unknown (bytes = "0F", bits = "1111"); Reference Station ID = Unknown (bytes = "F0 FF", bits = "111111111111"); Age of DGNSS Corrections = Unknown (bytes = "FF FF")
2022-09-28-11:36:59.668 3   0 255 129029 GNSS Position Data:  SID = 231 (bytes = "E7"); Date = 2013.03.01 (bytes = "95 3D"); Time = 19:29:52 (bytes = "00 73 D6 29"); Latitude = 42.4967684 (bytes = "00 DA 04 73 DB C9 E5 05"); Longitude = -71.5836637 (bytes = "80 7D 02 28 5F D6 10 F6"); Altitude = 90.984603 m (bytes = "9B 50 6C 05 00 00 00 00"); GNSS type = GPS+SBAS/WAAS (bytes = "03", bits = "0011"); Method = GNSS fix (bytes = "10", bits = "0001"); Integrity = No integrity checking (bytes = "00", bits = "00"); Number of SVs = 8 (bytes = "08"); HDOP = 1.11 (bytes = "6F 00"); PDOP = 1.90 (bytes = "BE 00"); Geoidal Separation = -33.63 m (bytes = "DD F2 FF FF"); Reference Stations = 0 (bytes = "00"); Reference Station Type 1 = Unknown (bytes = "0F", bits = "1111"); Reference Station ID = Unknown (bytes = "F0 FF", bits = "111111111111"); Age of DGNSS Corrections = Unknown (bytes = "FF FF")
